test/components/views/context_menus/MessageContextMenu-test.tsx

```diff
@@ -82,6 +74,103 @@ describe('MessageContextMenu', () => {
         expect(copyLinkButton).toHaveLength(0);
     });
 
+    describe('message forwarding', () => {
+        it('allows forwarding a room message', () => {
+            mocked(isContentActionable).mockReturnValue(true);
+
+            const eventContent = MessageEvent.from("hello");
+            const menu = createMenuWithContent(eventContent);
+            expect(menu.find('div[aria-label="Forward"]')).toHaveLength(1);
+        });
+
+        it('does not allow forwarding a poll', () => {
+            const eventContent = PollStartEvent.from("why?", ["42"], M_POLL_KIND_DISCLOSED);
+            const menu = createMenuWithContent(eventContent);
+            expect(menu.find('div[aria-label="Forward"]')).toHaveLength(0);
+        });
+
+        describe('forwarding beacons', () => {
+            const aliceId = "@alice:server.org";
+            beforeEach(() => {
+                mocked(isContentActionable).mockReturnValue(true);
+            });
+
+            it('does not allow forwarding a beacon that is not live', () => {
+                const deadBeaconEvent = makeBeaconInfoEvent(aliceId, roomId, { isLive: false });
+                const beacon = new Beacon(deadBeaconEvent);
+                const beacons = new Map<BeaconIdentifier, Beacon>();
+                beacons.set(getBeaconInfoIdentifier(deadBeaconEvent), beacon);
+                const menu = createMenu(deadBeaconEvent, {}, {}, beacons);
+                expect(menu.find('div[aria-label="Forward"]')).toHaveLength(0);
+            });
+
+            it('does not allow forwarding a beacon that is not live but has a latestLocation', () => {
+                const deadBeaconEvent = makeBeaconInfoEvent(aliceId, roomId, { isLive: false });
+                const beaconLocation = makeBeaconEvent(
+                    aliceId, { beaconInfoId: deadBeaconEvent.getId(), geoUri: 'geo:51,41' },
+                );
+                const beacon = new Beacon(deadBeaconEvent);
+                // @ts-ignore illegally set private prop
+                beacon._latestLocationEvent = beaconLocation;
+                const beacons = new Map<BeaconIdentifier, Beacon>();
+                beacons.set(getBeaconInfoIdentifier(deadBeaconEvent), beacon);
+                const menu = createMenu(deadBeaconEvent, {}, {}, beacons);
+                expect(menu.find('div[aria-label="Forward"]')).toHaveLength(0);
+            });
+
+            it('does not allow forwarding a live beacon that does not have a latestLocation', () => {
+                const deadBeaconEvent = makeBeaconInfoEvent(aliceId, roomId, { isLive: false });
```

Comment:
The test content does not match the description: „live“ ↔ `isLive: false` and it looks like having a latest location.

Can you check that?

Reply:
oops, too much copy pasting! fixed